api/provider/aws/environment_delete.go

```diff
@@ -51,32 +51,35 @@ func (e *EnvironmentProvider) Delete(environmentID string) error {
 		}
 
 		// Check for eventually consistency
-		fn := func() (shouldRetry bool, err error) {
+		var err error
+		fn := func() (shouldRetry bool) {
 			filter := &ec2.Filter{}
 			filter.SetName("group-name")
 			filter.SetValues([]*string{aws.String(securityGroupName)})
 
 			input := &ec2.DescribeSecurityGroupsInput{}
 			input.SetFilters([]*ec2.Filter{filter})
 
-			output, err := e.AWS.EC2.DescribeSecurityGroups(input)
+			var output *ec2.DescribeSecurityGroupsOutput
+			output, err = e.AWS.EC2.DescribeSecurityGroups(input)
 			if err != nil {
-				return false, err
+				return false
 			}
 
 			for _, group := range output.SecurityGroups {
 				if aws.StringValue(group.GroupName) == securityGroupName {
 					log.Printf("[DEBUG] Service group not deleted, will retry lookup")
-					return true, nil
+					err = errors.New(errors.EventualConsistencyError, err)
```

Comment:
err is `nil` here, so you need to create one to send to `errors.New` or use `errors.Newf`